src/porepy/grids/fv_sub_grid.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+import scipy.sparse as sps
+import porepy as pp
+
+
+class FvSubGrid(pp.Grid):
```

Comment:
I wrote this function exactly with the proper sub-grid in mind. In 2D it was straight-forward, but in 3D it was a bit more tricky, and I abounded this function and just forgot about it. 

I can not decide if it has the right of life or not. It might be a convenient way to set the subface-boundary conditions if you don't know anything about SubfaceTopology, however, then you are not able to interpret the output of Mpsa which is on the subface level. 

I think the cleanest would be to actually construct the sub-cell grid, let mpsa/mpfa work on this, and return the discretization on the sub-grid. This would simplify mpsa/mpfa considerably and possible make the code readable. But as you say, this is a project in its own